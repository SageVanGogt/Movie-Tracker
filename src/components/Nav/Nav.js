import { connect } from 'react-redux';
import React from 'react';
import { logOutUser} from '../../actions';
import {NavLink} from 'react-router-dom';

const Nav = (props) => {
  return (
    <div>
      { props.user.id ? 
        <button
        onClick={props.handleLogout}
        >Sign Out</button>
      :
      <NavLink to="/login">Login/Sign up</NavLink>
      }
    </div>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  handleLogout: () => dispatch(logOutUser())
})

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
