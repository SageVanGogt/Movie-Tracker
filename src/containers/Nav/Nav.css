a {
    color: #fff;
    margin-right: 60px;
    margin-top: 70px;
    text-decoration: none;
}

a:hover {
    border-bottom: solid 3px #fff;
    font-weight: 900;
}


.nav-bar {
    background-color: #a108ff;
    box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.40);
    display: flex;
    justify-content: flex-end;
    height: 100px;
    position: fixed;
    width: 100%;
    z-index: 1000;
}

.logo {
    width: 200px;
}