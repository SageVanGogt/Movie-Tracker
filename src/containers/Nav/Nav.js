import { connect } from 'react-redux';
import React from 'react';
import { logOutUser} from '../../actions';
import {NavLink} from 'react-router-dom';
import PropTypes from 'prop-types';
import './Nav.css';
import logo from '../../assets/logo.svg';


export const Nav = (props) => {
  const signOutBtn = (
    <NavLink to='/' 
      className="sign-out"
      onClick={props.handleLogout}>
      Sign Out
    </NavLink>
  );
  return (
    <div className="nav-bar">
      <img className="logo" src={logo} alt="movie tracker logo"/>
      <NavLink to="/favorites">Favorites</NavLink>
      <NavLink to="/">Home</NavLink>
      { 
        props.user.user_id ? 
        signOutBtn
        :
        <NavLink to="/login">Login</NavLink>
      }
    </div>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  handleLogout: () => dispatch(logOutUser())
})

export const mapStateToProps = (state) => ({
  user: state.user
})

Nav.propTypes = {
  handleLogout: PropTypes.func,
  user: PropTypes.object
}

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
