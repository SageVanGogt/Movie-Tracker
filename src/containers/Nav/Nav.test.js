import React from 'react';
import { Nav, mapDispatchToProps, mapStateToProps } from './Nav';
import {shallow} from 'enzyme'

describe('Nav', () => {

  it('should match the snapshot', () => {
    let mockProps = {
      user:{
        user_id:1
      }
    }
    let wrapper = shallow(<Nav {...mockProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch)
      const mockAction = {
        type: "LOGOUT_USER"
      }
      mappedProps.handleLogout()

      expect(mockDispatch).toHaveBeenCalledWith(mockAction)
    })
  })
  describe('mapStateToProps', () => {
    it('should map the user to props', () => {
      const mockState = {
        user: {
          id: 1,
          name: 'Alan',
          },
        films: []
      }

      const expected = {
        user: {
          id: 1,
          name: 'Alan',
        }
      }
      
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
})