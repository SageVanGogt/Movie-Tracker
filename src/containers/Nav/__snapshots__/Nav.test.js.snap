// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    user={
        Object {
            "user_id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="movie tracker logo"
          className="logo"
          src="logo.svg"
/>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/favorites"
>
          Favorites
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/"
>
          Home
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="sign-out"
          onClick={undefined}
          to="/"
>
          Sign Out
</NavLink>,
      ],
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie tracker logo",
          "className": "logo",
          "src": "logo.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Favorites",
          "to": "/favorites",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Home",
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Sign Out",
          "className": "sign-out",
          "onClick": undefined,
          "to": "/",
        },
        "ref": null,
        "rendered": "Sign Out",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="movie tracker logo"
            className="logo"
            src="logo.svg"
/>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/favorites"
>
            Favorites
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/"
>
            Home
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="sign-out"
            onClick={undefined}
            to="/"
>
            Sign Out
</NavLink>,
        ],
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie tracker logo",
            "className": "logo",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Favorites",
            "to": "/favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Home",
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Sign Out",
            "className": "sign-out",
            "onClick": undefined,
            "to": "/",
          },
          "ref": null,
          "rendered": "Sign Out",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
